[CI] Refactor CI e2e

lzhecheng opened the pull request:
* Use formats like "NS/Pod" in antreapolicy_test
* Use regexp to replace field value in mutateAntreaConfigMap()
* Remove unnecessary parameters in functions

lzhecheng commented:
/test-e2e

antoninbas commented on test/e2e/framework.go:
```diff
@@ -529,21 +534,20 @@ func (data *TestData) deployAntreaIPSec() error {
+		// We will skip TLS testing for Kind cluster because the server certificate is generated with Flow aggregator's DNS name
+		ap = append(ap, configPair{"enableTLSToFlowAggregator", "false"})
+	}
+	return data.mutateAntreaConfigMap([]configPair{}, ap, false, true)
```
nit: could use `nil` for the first argument

lzhecheng replied:
Updated.

antoninbas commented on test/e2e/framework.go:
```diff
@@ -1613,12 +1617,23 @@ func (data *TestData) GetGatewayInterfaceName(antreaNamespace string) (string, e
 }
 
-func (data *TestData) mutateAntreaConfigMap(mutatingFunc func(data map[string]string), restartController, restartAgent bool) error {
+func (data *TestData) mutateAntreaConfigMap(controllerPair []configPair, agentPair []configPair, restartController, restartAgent bool) error {
```
maybe `controllerChanges` and `agentChanges` would be better names than `controllerPair` and `agentPair`, and I feel like `configChange` is a better name than `configPair` for the type

lzhecheng replied:
Updated.

antoninbas commented on test/e2e/framework_test.go:
```diff
@@ -0,0 +1,36 @@
+	"testing"
+)
+
+func TestReplaceFieldValue(t *testing.T) {
```
It's a bit strange that this will be run as part of the e2e tests, but I don't have a better idea and since it completes in an instant, it's not really an issue

lzhecheng replied:
Yes, test code needs testing.

tnqn replied:
I found this test when looking at several test failures and it happened to be the last succeeded e2e test in half cases. Although it didn't take me much time to exclude it as a suspect, it was distracting to debug a unit test in e2e test suite. Does it really help to test testing code? If the function is broken, won't the target test fail?

antoninbas replied:
I have an idea. We could add a build tag to this test file. The e2e test suite will no longer run the test, but we can run it as part of the regular Go unit testing Github workflow using a command like this one:

```
go test --tags=e2e-utest github.com/vmware-tanzu/antrea/test/e2e/...
```

antoninbas commented on test/e2e/k8s_util.go:
```diff
@@ -48,7 +48,7 @@ func NewKubernetesUtils(data *TestData) (*KubernetesUtils, error) {
 	pods, err := k.getPodsUncached(ns, "pod", name)
 	if err != nil {
-		return nil, errors.WithMessagef(err, "unable to get Pod is ns %s with label pod=%s", ns, name)
+		return nil, errors.WithMessagef(err, "unable to get Pod in namespace %s with label pod=%s", ns, name)
```
s/namespace/Namespace

lzhecheng replied:
Updated.

antoninbas reviewed (commented):
couple nits, looks there are some build errors

lzhecheng commented:
/test-e2e

antoninbas reviewed (approved):
LGTM

lzhecheng commented:
/test-e2e

lzhecheng commented:
/test-e2e

lzhecheng commented:
@antoninbas I just fixed some problems and rebased the PR due to conflict with main. Now Kind and Jenkins e2e test are successful. Could you please review again? Thank you!

lzhecheng commented:
/skip-conformance
/skip-networkpolicy
